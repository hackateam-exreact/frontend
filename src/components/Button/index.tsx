import {
  Button as ChakraButton,
  ButtonProps as ChakraButtonProps
} from '@chakra-ui/react'

export interface ButtonProps extends ChakraButtonProps {
  backgroundColor?: 'blue.500' | 'green.500'
  children: string
  fullWidth?: boolean
  size?: 'xs' | 'sm' | 'md' | 'lg'
}

export const Button = ({
  backgroundColor = 'blue.500',
  children,
  fullWidth = false,
  size = 'md',
  ...props
}: ButtonProps) => (
  <ChakraButton
    color="white"
    backgroundColor={backgroundColor}
    width={fullWidth ? '100%' : 'auto'}
    size={size}
    _hover={{
      backgroundColor
    }}
    {...props}
  >
    {children}
  </ChakraButton>
)
