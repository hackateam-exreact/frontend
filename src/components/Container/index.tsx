import { Flex, FlexProps, useColorMode } from '@chakra-ui/react'

export const Container = (props: FlexProps) => {
  const { colorMode } = useColorMode()

  const bgColor = { light: 'gray.50', dark: 'black.500' }
  const color = { light: 'black', dark: 'white.500' }

  return (
    <Flex
      direction="column"
      alignItems="center"
      justifyContent="flex-start"
      bg={bgColor[colorMode]}
      color={color[colorMode]}
      {...props}
    />
  )
}
