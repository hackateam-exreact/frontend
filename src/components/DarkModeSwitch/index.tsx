import { useColorMode, Switch } from '@chakra-ui/react'

const DarkModeSwitch = () => {
  const { colorMode, toggleColorMode } = useColorMode()
  const isDark = colorMode === 'dark'

  return (
    <Switch
      position="fixed"
      top="1rem"
      right="1rem"
      aria-label="toggle color mode"
      isChecked={isDark}
      onChange={toggleColorMode}
    />
  )
}

export default DarkModeSwitch
