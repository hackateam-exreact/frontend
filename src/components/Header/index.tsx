import { Link as ChakraLink, Flex, HStack, Image } from '@chakra-ui/react'

import Link from 'next/link'
import React from 'react'
import { Searchbar } from './Searchbar'
import { UserBadge } from 'components/Profile/UserBadge'
import { useAuth } from 'hooks/useAuth'
import { useRouter } from 'next/router'

export function Header() {
  const { user } = useAuth()
  const { asPath } = useRouter()

  const blacklist = ['/', '/dev/signin', '/dev/signup', '/dev/welcome']

  return (
    <>
      {!blacklist.includes(asPath) && (
        <Flex
          w="100%"
          h="6rem"
          borderBottom="1px"
          borderColor="gray.800"
          mb="1rem"
        >
          <HStack
            w="100%"
            maxW="1400px"
            mx="auto"
            align="center"
            spacing="10"
            px="5"
          >
            <Link href="/" passHref={true}>
              <ChakraLink>
                <Image
                  src="/img/logo.png"
                  alt="Devspot"
                  w="225px"
                  objectFit="cover"
                />
              </ChakraLink>
            </Link>
            <Searchbar />
            <UserBadge user={user} badgeSide="right" />
          </HStack>
        </Flex>
      )}
    </>
  )
}
