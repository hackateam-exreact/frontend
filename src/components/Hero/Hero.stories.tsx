import { Story, Meta } from '@storybook/react/types-6-0'

import { Hero, HeroProps } from '.'

export default {
  title: 'Hero',
  component: Hero,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: Story<HeroProps> = (args) => <Hero {...args} />
