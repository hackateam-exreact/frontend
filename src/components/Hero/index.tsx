import { Flex, Heading } from '@chakra-ui/react'

export type HeroProps = {
  title?: string
}

export const Hero = ({ title = 'Fullstack Alchemists' }: HeroProps) => (
  <Flex
    justifyContent="center"
    alignItems="center"
    height="100vh"
    bgGradient="linear(to-l, #7928CA, #FF0080)"
    bgClip="text"
  >
    <Heading fontSize="6vw">{title}</Heading>
  </Flex>
)
