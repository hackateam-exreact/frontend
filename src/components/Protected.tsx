import { ReactNode, useEffect, useState } from 'react'

import { useAuth } from 'hooks/useAuth'

interface ProtectedProps {
  needsAuthentication?: boolean
  needsAuthorization?: boolean
  children: ReactNode
}

export function Protected({
  needsAuthentication = true,
  needsAuthorization = false,
  children
}: ProtectedProps) {
  const { isAuthenticated, isAuthorized } = useAuth()
  const [canShow, setCanShow] = useState(false)

  const handleCheckPermissions = () => {
    if (needsAuthentication && needsAuthorization) {
      if (isAuthenticated && isAuthorized) return true
    }

    if (needsAuthentication) {
      if (isAuthenticated) return true
    }

    if (needsAuthorization) {
      if (isAuthorized) return true
    }

    return false
  }

  useEffect(() => {
    setCanShow(handleCheckPermissions())
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isAuthenticated, isAuthorized])

  return <>{canShow && children}</>
}
