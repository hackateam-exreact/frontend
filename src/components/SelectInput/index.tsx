import {
  Select as ChakraSelect,
  FormControl,
  FormErrorMessage,
  FormLabel,
  SelectProps
} from '@chakra-ui/react'
import { ForwardRefRenderFunction, forwardRef } from 'react'

import { FieldError } from 'react-hook-form'
import { ReactNode } from 'react'

export interface SelectInputPropt extends SelectProps {
  children: ReactNode
  inputName: string
  label?: string
  error?: FieldError
  focusBorderColor?: 'blue.500' | 'green.500'
}

const SelectInputBase: ForwardRefRenderFunction<
  HTMLSelectElement,
  SelectInputPropt
> = (
  {
    children,
    inputName,
    label,
    error = null,
    focusBorderColor = 'blue.500',
    ...rest
  },
  ref
) => {
  return (
    <FormControl isInvalid={!!error}>
      {label && <FormLabel htmlFor={inputName}>{label}</FormLabel>}
      <ChakraSelect
        name={inputName}
        id={inputName}
        focusBorderColor={focusBorderColor}
        errorBorderColor="red.500"
        bg="gray.800"
        border="none"
        size="lg"
        _placeholder={{ fontSize: 'sm' }}
        ref={ref}
        {...rest}
      >
        {children}
      </ChakraSelect>
      {!!error && (
        <FormErrorMessage color="red.500">{error.message}</FormErrorMessage>
      )}
    </FormControl>
  )
}

export const SelectInput = forwardRef(SelectInputBase)
