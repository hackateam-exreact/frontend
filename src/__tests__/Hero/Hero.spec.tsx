import { render, screen } from 'utils/test-utils'

import { Hero } from '../../components/Hero'

describe('<Hero/ >', () => {
  it('should render with a default heading', () => {
    const { container } = render(<Hero />)

    expect(
      screen.getByRole('heading', {
        name: /fullstack alchemists/i
      })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render with a passed heading', () => {
    const newTitle = 'testing title'
    render(<Hero title={newTitle} />)

    expect(screen.getByRole('heading', { name: newTitle })).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: newTitle }).parentElement
    ).toHaveStyleRule(
      'background-image',
      'linear-gradient(to left, #7928CA, #FF0080)'
    )
  })
})
