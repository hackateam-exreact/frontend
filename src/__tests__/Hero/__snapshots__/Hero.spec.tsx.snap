// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero/ > should render with a default heading 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100vh;
  background-image: linear-gradient(to left, #7928CA, #FF0080);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

.emotion-1 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: 6vw;
  line-height: 1.33;
}

@media screen and (min-width: 48em) {
  .emotion-1 {
    line-height: 1.2;
  }
}

<body
  class="chakra-ui-dark"
>
  <div
    id="chakra-toast-portal"
  >
    <ul
      id="chakra-toast-manager-top"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-top-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-top-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-bottom-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
  <div>
    <div
      class="emotion-0"
    >
      <h2
        class="chakra-heading emotion-1"
      >
        Fullstack Alchemists
      </h2>
    </div>
  </div>
</body>
`;
